src/main/java/com/shaft/validation/internal/ValidationsHelper2.java

```diff
@@ -75,6 +86,11 @@ protected void validateBrowserAttribute(WebDriver driver, String attribute,
 
         try {
             new SynchronizationManager(driver).fluentWait(false).until(f -> {
+                // Increase the bar after 6 (100) millisecond runs to take into account code execution time
+                if(millisCount.incrementAndGet() == 6){
```

Comment:
this is the root cause for the race condition issue (sometimes the counter ends at 80 or 90% not reaching 100%)
instead of creating the logic inside the fluent wait block, it should be created outside it and run in a separate virtual thread.

it is recommended to implement this virtual thread outside this class to avoid code duplication, specifically in the ValidationsExecutor class (which calls the validation helper) before the switch case statement that determines how the assertion will be performed.

Reply:
* Thanks for the virtual thread suggestion, it is working perfectly 😄. 
* Regarding calling the thread before the switch case in the `ValidationsExecutor` class, I tried this approach, but the problem is that the progress bar is called multiple times during which it should not be called as shown in the screenshot below. Thus causes the console log to appear distorted. 
* The progress bar is made specifically to be called only during a fluent wait so that it shows the user that the script is actually running and waiting for the assertion to pass thus preventing the "false stuck script" issue.

![Screenshot](https://github.com/ShaftHQ/SHAFT_ENGINE/assets/37153097/5c4d605a-cc83-4ea6-bf55-10e312279f02)